modules/javafx.controls/src/main/java/javafx/scene/control/TableUtil.java

```diff
@@ -223,7 +223,7 @@ static boolean constrainedResize(ResizeFeaturesBase prop,
             colWidth += col.getWidth();
         }
 
-        if (Math.abs(colWidth - tableWidth) > 1) {
+        if (Math.abs(colWidth - tableWidth) > 0) {
```

Comment:
Related to Andy's question, shouldn't this be `> EPSILON` (for some small value of EPSILON), since floating-point arithmetic isn't precise?

Reply:
@kevinrushforth We can consider EPSILON with value of `.0000001`. This value is already used in the same method for another bounds checking. Below is the part of the code where it is used..
```
// Check for zero. This happens when the distribution of the delta
// finishes early due to a series of "fixed" entries at the end.
// In this case, lowerBound == upperBound, for all subsequent terms.
double newSize;
if (Math.abs(totalLowerBound - totalUpperBound) < .0000001) {
    newSize = lowerBound;
} else {
    double f = (target - totalLowerBound) / (totalUpperBound - totalLowerBound);
    newSize = Math.round(lowerBound + f * (upperBound - lowerBound));
}
```

Reply:
Just a question out of curiosity, is there some kind of recommended value for epsilon? 
I remember we also have very small epsilon in `Region` for snapping.

Reply:
This is a *good* question.
In this case, we are summing column widths, so we can estimate the worst case scenario by simply adding max errors accumulated after each addition:
```
sum(N` * 2 * ε * |w|)
```
where
N = number of columns
ε = upper bound for rounding error, or epsilon which is about 1e-16 [0]
w = column width

So, plugging a 100 columns of 200 px wide we get something to the order of 1e-12 .. 1e-11.

On a side note, we could have used Unicode here
```
private static final double ε = .0000001;
```

[0] https://en.wikipedia.org/wiki/Machine_epsilon
[1] https://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html

Reply:
> On a side note, we could have used Unicode here
> 
> private static final double ε = .0000001;

But let's not.

Reply:
you are right, we should not use non-ascii identifiers.
it was a joke, sorry.

Reply:
> So, plugging a 100 columns of 200 px wide we get something to the order of 1e-12 .. 1e-11.

Thanks for the explaination and links.
The remaining question is: Should we then use a smaller epsilon value or does it not matter at all?

Reply:
In general, the smallest epsilon depends on the magnitude of the floating-point number. There’s `Math.ulp(double)`, which for a given double value, returns the distance to the next representable value.